src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentRenderModeDirectiveIntegrationTests.cs

```diff
@@ -0,0 +1,150 @@
+    }
+
+    internal override CSharpCompilation BaseCompilation { get; }
+
+    internal override string FileKind => FileKinds.Component;
+
+    [Fact]
+    public void RenderMode_With_Fully_Qualified_Type()
+    {
+        // Arrange & Act
+        var component = CompileToComponent($@"
+@rendermode Microsoft.AspNetCore.Components.DefaultRenderModes.Server
+");
+
+        // Assert
+        var attribute = Assert.Single(component.GetType().CustomAttributes);
+        Assert.Equal("PrivateComponentRenderModeAttribute", attribute.AttributeType.Name);
+
+        var nestedType = component.GetType().GetNestedType("PrivateComponentRenderModeAttribute", System.Reflection.BindingFlags.NonPublic);
+        Assert.NotNull(nestedType);
+
+        var modeProperty = nestedType.GetProperty("Mode");
+        Assert.NotNull(modeProperty);
+    }
+
+    [Fact]
+    public void RenderMode_With_Static_Usings()
+    {
+        // Arrange & Act
+        var component = CompileToComponent($@"
+@using static Microsoft.AspNetCore.Components.DefaultRenderModes
+@rendermode Server
+");
+
+        // Assert
+        var attribute = Assert.Single(component.GetType().CustomAttributes);
+        Assert.Equal("PrivateComponentRenderModeAttribute", attribute.AttributeType.Name);
+
+        var nestedType = component.GetType().GetNestedType("PrivateComponentRenderModeAttribute", System.Reflection.BindingFlags.NonPublic);
+        Assert.NotNull(nestedType);
+
+        var modeProperty = nestedType.GetProperty("Mode");
+        Assert.NotNull(modeProperty);
+    }
+
+    [Fact]
+    public void RenderMode_Missing_Value()
+    {
+        // Arrange & Act
+        var compilationResult = CompileToCSharp($@"
+@rendermode
+");
+
+        // Assert
+        // Error RZ1014: The 'rendermode' directive expects a namespace name.
+        var diagnostic = Assert.Single(compilationResult.Diagnostics);
+        Assert.Equal("RZ1014", diagnostic.Id);
+    }
+
+    [Fact]
+    public void DuplicateRenderModes()
+    {
+        // Arrange & Act
+        var compilationResult = CompileToCSharp($@"
+@rendermode Microsoft.AspNetCore.Components.DefaultRenderModes.Server
+@rendermode Microsoft.AspNetCore.Components.DefaultRenderModes.Server
+");
+
+        // Assert
+        //Error RZ2001: The 'rendermode' directive may only occur once per document.
+        var diagnostic = Assert.Single(compilationResult.Diagnostics);
+        Assert.Equal("RZ2001", diagnostic.Id);
+    }
+
+    [Fact]
+    public void RenderMode_With_InvalidIdentifier()
+    {
+        var compilationResult = CompileToCSharp($@"
+@rendermode NoExist
+");
+
+        Assert.Empty(compilationResult.Diagnostics);
+
+        var assemblyResult = CompileToAssembly(compilationResult, throwOnFailure: false);
+        assemblyResult.Diagnostics.Verify(
+                    // x:\dir\subdir\Test\TestComponent.cshtml(1,58): error CS0103: The name 'NoExist' does not exist in the current context
```

Comment:
I know this was talked about at the LDM meeting, but I definitely think in general we should prioritize good diagnostics from the compiler if possible. Even something like "Expected identifier or expression of type IComponentRenderMode" would be infinitely more helpful to users, as it would give them something good to google.

Reply:
Agree. I'm going to open an issue to track better error handling though rather than adding it to this PR.